server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -491,6 +491,13 @@ public Map<String, Object> getSourceAsMap() {
         return sourceAsMap;
     }
 
+    /**
+     * Set the cache document as a map to {@code null}.
+     */
+    public void resetSourceAsMap() {
```

Comment:
Good idea! Shall we also add this to `SearchHit#deallocate`?

Reply:
Sure, I nullify it in `SearchHit#deallocate` too

Reply:
Ah sorry for some reason I missed that line :man_facepalming: don't mind me :)